gateway/src/main/java/io/zeebe/gateway/EndpointManager.java

```diff
@@ -400,32 +401,36 @@ public static StatusRuntimeException convertThrowable(final Throwable cause) {
 
     if (cause instanceof BrokerErrorException) {
       status = mapBrokerErrorToStatus(((BrokerErrorException) cause).getError());
+      // When there is back pressure, there will be a lot of `RESOURCE_EXHAUSTED` errors and the log
+      // can get flooded. Until we find a way to limit the number of log messages,
+      // let's do not log them.
+      if (status.getCode() != Status.RESOURCE_EXHAUSTED.getCode()) {
+        Loggers.GATEWAY_LOGGER.error(GRPC_ERROR_TEMPLATE + "received error from broker", cause);
+      }
     } else if (cause instanceof BrokerRejectionException) {
       status = mapRejectionToStatus(((BrokerRejectionException) cause).getRejection());
+      Loggers.GATEWAY_LOGGER.trace(GRPC_ERROR_TEMPLATE + "broker rejected request", cause);
     } else if (cause instanceof ClientOutOfMemoryException) {
       status = Status.UNAVAILABLE.augmentDescription(cause.getMessage());
+      Loggers.GATEWAY_LOGGER.error(GRPC_ERROR_TEMPLATE + "gateway out of memory", cause);
```

Comment:
I looked into it, and this case I think is not possible anymore - a `ClientOutOfMemoryException` would happen literally only if the JVM ran out of memory, which would already throw an OOM, so I don't think this exception is even used anymore?

Reply:
That could very well be. It is thrown in the `BrokerRequestManager` in the case that the future is null, which seemed strange to me. Is that how we would normally detect an OOM?

Reply:
No, it's a historical remnant from when we had a queue where you would add your requests. If that queue was full, then you would not get a future back - but we don't have this anymore, so I don't expect it ever happens. I'd remove it, though I'm also OK if you add a separate issue for that.